pkg/sfu/connectionquality/connectionstats.go

```diff
@@ -118,11 +119,18 @@ func (cs *ConnectionStats) updateScore(streams map[uint32]*buffer.StreamStatsWit
 			params.IsReducedQuality = cs.params.GetIsReducedQuality()
 		}
 		cs.score = ScreenshareTrackScore(params)
+
 	case cs.params.CodecType == webrtc.RTPCodecTypeAudio:
-		if cs.params.IsDtxDisabled != nil {
-			params.DtxDisabled = cs.params.IsDtxDisabled()
+		packetRate := float64(params.PacketsExpected) / maxAvailableLayerStats.Duration.Seconds()
+		if packetRate > audioPacketRateThreshold {
+			// With DTX, it is possible to have fewer packets per second.
+			// A loss with reduced packet rate has amplified effect on uality.
+			// Opus uses 20 ms packetisation (50 pps). Calculate score only if packet rate is at least half of that.
+			if cs.params.IsDtxDisabled != nil {
+				params.DtxDisabled = cs.params.IsDtxDisabled()
+			}
+			cs.score = AudioTrackScore(params)
```

Comment:
Typically no. `PacketsExpected` is based on sequence number difference (last sn - first sn + 1 in the window). So, if the loss is > 50%, the packets expected would still exceed the rate.

But, in extreme cases, it can skip the score calculation.  If we get 50 packets in the first `second` of the window and then nothing in the second `second` of the window. That will be 25 pps rate and 0% packet loss. As there are no packets in the second `second`, last sn is what we received and expected packets would still be 50 and hence the loss is 0%. In that case, we will not update the score. Even if we receive 51 packets, we will calculate the score using 0% loss. Will think about such cases. Currently, when there are no packets expected, we skip analytics posting and scoring. So, if that stoppage is prolonged, we will not catch it as a drop in score. We might need some checks for presence of traffic in the quality calculation.